visidata/_open.py

```diff
@@ -81,6 +81,10 @@ def guess_extension(vd, path):
 def openPath(vd, p, filetype=None, create=False):
     '''Call ``open_<filetype>(p)`` or ``openurl_<p.scheme>(p, filetype)``.  Return constructed but unloaded sheet of appropriate type.
     If True, *create* will return a new, blank **Sheet** if file does not exist.'''
+    # allow user to assign a filetype to a pathname:  options.set('filetype', 'csv', '-')
+    filetype = filetype or vd.options.getonly('filetype', str(p), None)  #1710
+    filetype = filetype or vd.options.getonly('filetype', 'global', None)
```

Comment:
Why were these lines moved?  It seems like code at line 102/103 should fix the issue by itself.

Reply:
The code at line 102 would fix it by itself. Lines 85, 86, and 104 are all similar in filling in filetype based on options. The intention was to put them together in one function, to make it easier to trace the logic that sets filetype based on options.